pkg/utils/pod/scheduling.go

```diff
@@ -45,6 +45,29 @@ func IsSchedulable(pod *v1.PodSpec, node *v1.Node) bool {
 	return true
 }
 
+// IgnoredForUnderutilization returns true if the set of pods has no non-daemonset pods
+func IgnoredForUnderutilization(pods []*v1.Pod) bool {
+	for _, p := range pods {
+		if HasFailed(p) {
+			continue
+		}
+		if !IsOwnedByDaemonSet(p) {
+			return false
+		}
+	}
+	return true
+}
+
+// AreAllTerminating returns true if the set of pods were all unable to join
+func AreAllTerminating(pods []*v1.Pod) bool {
```

Comment:
Is this function the same as IgnoredForTermination or something different?

Reply:
IgnoredForTermination feels like we're saying we want to terminate these pods, while in reality they are already terminating. I like this name more.

Reply:
Sorry my typo was misleading. I meant:
>Is this function the same as IgnoredForUnderutilizationor something different?

What I mean is, the logic looks very similar to me. The only difference is that they behave differently w/ deletion timestamp. They both should do the same thing for daemonsets and failed pods.